src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -144,14 +144,28 @@ public static string GetEffectiveConfigAsString(string gitDirectory, IFileSystem
             return stringBuilder.ToString();
         }
 
-        static string GetConfigFilePath(string workingDirectory)
+        static string GetConfigFilePath(string workingDirectory, IFileSystem fileSystem)
         {
-            return Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            var ymlPath = Path.Combine(workingDirectory, "GitVersion.yml");
+            if (fileSystem.Exists(ymlPath))
+            {
+                return ymlPath;
+            }
+
+            var yamlPath = Path.Combine(workingDirectory, "GitVersionConfig.yaml");
+            if (fileSystem.Exists(yamlPath))
+            {
+                Logger.WriteWarning("'GitVersionConfig.yaml' is deprecated, use 'GitVersion.yml' instead.");
+
+                return yamlPath;
+            }
+
+            return ymlPath;
```

Comment:
@qetza How about renaming the variable `yamlPath` to `deprecatedPath`? Just a single letter separating two variables make them, as obvious by @pascalberger's comment, hard to differentiate. :smiley: